src/measure_theory/interval_integral.lean

```diff
@@ -0,0 +1,274 @@
+  norm_integral_eq_norm_integral_Ioc
+... ≤ ∫ x in Ioc (min a b) (max a b), ∥f x∥ ∂μ :
+  norm_integral_le_integral_norm f
+
+lemma norm_integral_le_abs_integral_norm : ∥∫ x in a..b, f x ∂μ∥ ≤ abs (∫ x in a..b, ∥f x∥ ∂μ) :=
+begin
+  simp only [← real.norm_eq_abs, norm_integral_eq_norm_integral_Ioc],
+  exact le_trans (norm_integral_le_integral_norm _) (le_abs_self _)
+end
+
+lemma norm_integral_le_of_norm_le_const_ae {a b C : ℝ} {f : ℝ → E}
+  (h : ∀ᵐ x, x ∈ Ioc (min a b) (max a b) → ∥f x∥ ≤ C) :
+  ∥∫ x in a..b, f x∥ ≤ C * abs (b - a) :=
+begin
+  rw [norm_integral_eq_norm_integral_Ioc],
+  have : volume (Ioc (min a b) (max a b)) = ennreal.of_real (abs (b - a)),
+  { rw [real.volume_Ioc, max_sub_min_eq_abs, ennreal.of_real] },
+  rw [← ennreal.to_real_of_real (abs_nonneg _), ← this],
+  refine norm_set_integral_le_of_norm_le_const_ae'' _ is_measurable_Ioc h,
+  simp only [this, ennreal.lt_top_iff_ne_top],
+  exact ennreal.of_real_ne_top
+end
+
+lemma norm_integral_le_of_norm_le_const {a b C : ℝ} {f : ℝ → E}
+  (h : ∀ x ∈ Ioc (min a b) (max a b), ∥f x∥ ≤ C) :
+  ∥∫ x in a..b, f x∥ ≤ C * abs (b - a) :=
+norm_integral_le_of_norm_le_const_ae $ eventually_of_forall h
+
+lemma integral_add (hfm : measurable f) (hfi : interval_integrable f μ a b)
+  (hgm : measurable g) (hgi : interval_integrable g μ a b) :
+  ∫ x in a..b, f x + g x ∂μ = ∫ x in a..b, f x ∂μ + ∫ x in a..b, g x ∂μ :=
+begin
+  simp only [interval_integral, integral_add hfm hfi.1 hgm hgi.1,
+    integral_add hfm hfi.2 hgm hgi.2],
+  abel
+end
+
+@[simp] lemma integral_neg : ∫ x in a..b, -f x ∂μ = -∫ x in a..b, f x ∂μ :=
+begin
+  simp only [interval_integral, integral_neg],
+  abel
+end
+
+variables [topological_space α] [opens_measurable_space α]
+
+section order_closed_topology
+
+variables [order_closed_topology α]
+
+lemma integral_cocycle (hfm : measurable f) (hab : interval_integrable f μ a b)
+  (hbc : interval_integrable f μ b c) :
+  ∫ x in a..b, f x ∂μ + ∫ x in b..c, f x ∂μ + ∫ x in c..a, f x ∂μ = 0 :=
+begin
+  have hac := hab.trans hbc,
+  simp only [interval_integral, ← add_sub_comm, sub_eq_zero],
+  iterate 4 { rw ← integral_union },
+  { suffices : Ioc a b ∪ Ioc b c ∪ Ioc c a = Ioc b a ∪ Ioc c b ∪ Ioc a c, by rw this,
+    rw [Ioc_union_Ioc_union_Ioc_cycle, union_right_comm, Ioc_union_Ioc_union_Ioc_cycle,
+      min_left_comm, max_left_comm] },
+  all_goals { simp [*, is_measurable.union, is_measurable_Ioc, Ioc_disjoint_Ioc_same,
+    Ioc_disjoint_Ioc_same.symm, hab.1, hab.2, hbc.1, hbc.2, hac.1, hac.2] }
+end
+
+lemma integral_add_adjacent_intervals (hfm : measurable f) (hab : interval_integrable f μ a b)
+  (hbc : interval_integrable f μ b c) :
+  ∫ x in a..b, f x ∂μ + ∫ x in b..c, f x ∂μ = ∫ x in a..c, f x ∂μ :=
+by rw [← add_neg_eq_zero, ← integral_symm, integral_cocycle hfm hab hbc]
+
+end order_closed_topology
+
+variables [order_topology α]
+
+open asymptotics
+
+lemma integral_sub_linear_is_o_of_tendsto_ae [locally_finite_measure μ] {f : α → E} {a : α} {c : E}
```

Comment:
This lemma and the next two ones would deserve docstrings. Although, it feels a little strange that you don't give the statements assuming that `f` is continuous at `a`. There are also countless possible variations (for instance assuming that `ha` only holds on the right of `a`, and the conclusion would only be on the right of `a`, getting rid of `min` and `max` and so on), but these can wait for when they are needed, in a further PR.

Reply:
I give more versions of FTC and related statements in the `FTC-strict` branch. Backporting them here would mean backporting most of the diff.

Reply:
ok, so let's not change anything here, and postpone this to the next PR.